cmake/Platform/Initialization/SetDefaults.cmake

```diff
@@ -1,4 +1,5 @@
 set(ARDUINO_DEFAULT_BOARD uno CACHE STRING "Default Arduino Board ID when not specified.")
+set(ARDUINO_DEFAULT_BOARD_CPU CACHE STRING "Default Arduino Board CPU when not specified.")
```

Comment:
Seems like you didn't set a default after all - Do you expect the build to just fail in such case?

Reply:
Uhm, very good point... i should have tested the different combinations :(
Having a "fixed" default does not make sense, it should take the (only) cpu of the given/default board if unset. Will take a look at how to achieve that. (or feel free to fix if it is obvious for you)

Reply:
No, please, take a shot at it. I currently don't have the time to address it, plus, it seems like a small fix that should be applied, so it's a great place to start contributing 😃

Reply:
I just checked, when a board does have a single cpu (ie, no cpus defined) the value is simply ignored. So the build will only fail if ARDUINO_DEFAULT_BOARD is some board with multiple CPUs (e.g. diecimila) and ARDUINO_DEFAULT_BOARD_CPU is unset: but this is OK/expected i would say. If ARDUINO_DEFAULT_BOARD=uno, then it works fine.
So it doesn't really add anything IMHO to have e.g.  ARDUINO_DEFAULT_BOARD_CPU=atmega328p.

Anyways, i've now added a check to avoid setting INPUT_BOARD_CPU=ARDUINO_DEFAULT_BOARD_CPU if ARDUINO_DEFAULT_BOARD_CPU is unset. Due to the above, it shouldn't make any practical difference, but is somewhat cleaner:

> if (NOT INPUT_BOARD_CPU AND ARDUINO_DEFAULT_BOARD_CPU)
>     set(INPUT_BOARD_CPU ${ARDUINO_DEFAULT_BOARD_CPU})
> endif ()

What do you think?

Reply:
Thanks for researching the issue, it's good to know that boards with single CPUs are ignored.

This change looks great, though there is one more thing that I would add before completing it.
In case a board with multiple CPUs is used, no `BOARD_CPU` is defined and no `DEFAULT_BOARD_CPU` is defined, the build will fail without informing the user why. I would like to somehow fail the build using CMake's [message](https://cmake.org/cmake/help/latest/command/message.html?highlight=message) function, intended specifically for such purposes.
However, I also realize that doing so will require us to know in advance if the used board requires a specific CPU definition, which is impractical by any chance.

What do you think we should do about it?

Reply:
Hi thanks for the feedback.

No, if BOARD_CPU is missing for a BOARD with multiple cpu's you'll get:

>
> CMake Error at /home/ariel/soft/bin/packages/arduino-cmake/cmake/Platform/Core/BoardPropertiesReader.cmake:23 (message):
>   Board has multiple CPU versions (atmega2560;atmega1280).  BOARD_CPU must be
>   defined for target blink.
> Call Stack (most recent call first):
> ...
>

This was already the case earlier :) We could improve that message telling the user how to exactly set BOARD_CPU or pointing it at ARDUINO_DEFAULT_BOARD_CPU, although it may get a bit longish...

Reply:
Oh, indeed, my mistake...
Well, PR approved then 😄